samtranslator/validator/resource_validator.py

```diff
@@ -0,0 +1,47 @@
+"""A resource validator to help validate resource properties and raise exception when some value is unexpected."""
+from typing import Any, Dict, Type
+
+from pydantic import BaseModel
+
+
+class Model:
+    """
+    Wrapper class around a SAM schema BaseModel to with a new functional "get" method
+    """
+
+    def __init__(self, model: BaseModel) -> None:
+        self.model = model
+
+    def _process_attr_value(self, attr_value: Any) -> Any:
+        if isinstance(attr_value, BaseModel):
+            if "__root__" in attr_value.__dict__:
```

Comment:
Why? Comment with references help.

Reply:
This feels like we are using `BaseModel`'s internal implementation logic here. They may change it without making an announcement and our code might break.

Reply:
(as `__root__` is a private attribute of `BaseModel`)

Reply:
This is actually for PassThroughProperty we defined. Will add a comment.
```
class PassThroughProp(pydantic.BaseModel):
    __root__: PassThrough
```